host/windows/syscall.c

```diff
@@ -3124,3 +3124,39 @@ int oe_syscall_nanosleep_ocall(struct oe_timespec* req, struct oe_timespec* rem)
     _set_errno(0);
     return 0;
 }
+
+/*
+**==============================================================================
+**
+** clock_nanosleep():
+**
+**==============================================================================
+*/
+
+int oe_syscall_clock_nanosleep_ocall(
+    oe_clockid_t clockid,
+    int flag,
+    struct oe_timespec* req,
+    struct oe_timespec* rem)
+{
+    /* Validate input */
+    if (flag != 0)
+    {
+        /*
+         * In Windows, we do not support the feature to mention a future
+         * absolute time on the clock upto which to sleep (ie; flag
+         * TIMER_ABSTIME is not supported).
+         */
+        _set_errno(OE_EINVAL);
+        return -1;
+    }
+    /*
+     * Choosing a clock is not supported in Windows. It is safe to ignore
+     * a request to use a specific clock, as the flag TIMER_ABSTIME
+     * is not supported and the sleep duration units are the same for all
+     * available clocks.
+     */
+    OE_UNUSED(clockid);
```

Comment:
https://man7.org/linux/man-pages/man2/clock_nanosleep.2.html explains:
```
       EINVAL clockid was invalid.  (CLOCK_THREAD_CPUTIME_ID is not a
              permitted value for clockid.)
```
Similarly if people pass in other undefined values that would fail on Linux as being unsupported values, it would seem bug-prone for Windows to allow them rather than adhering to the contract in the man page for clock_nanosleep().

Reply:
Ah makes sense. I'll add checks similar to Linux. Thank you, Dave.

Reply:
Added checks to host/windows/syscall.c:oe_syscall_clock_nanosleep_ocall() as per https://code.woboq.org/userspace/glibc/sysdeps/unix/sysv/linux/clock_nanosleep.c.html